ui/src/main/java/org/pentaho/di/ui/trans/steps/fileinput/text/TextFileCSVImportProgressDialog.java

```diff
@@ -275,8 +276,15 @@ private String doScan( IProgressMonitor monitor, final boolean failOnParseError
     if ( meta.content.header ) {
       int skipped = 0;
       while ( line != null && skipped < meta.content.nrHeaderLines ) {
-        line = TextFileInputUtils.getLine( log, reader, encodingType, fileFormatType, lineBuffer );
-        skipped++;
+        if ( TextFileInputUtils.checkPattern( line, meta.getEnclosure() ) % 2 == 0 ) {
+          line = TextFileInputUtils.getLine( log, reader, encodingType, fileFormatType, lineBuffer );
+          skipped++;
+        } else {
+          sbLine.append( line );
+          line = sbLine.append(
+            TextFileInputUtils.getLine( log, reader, encodingType, fileFormatType, lineBuffer ) ).toString();
+          sbLine = new StringBuilder();
```

Comment:
Using a `StringBuilder` to concatenate two strings and then creating a new `StringBuilder` doesn't look very efficient in my book.